merkle/compact/tree.go

```diff
@@ -111,25 +122,41 @@ func NewTreeWithState(hasher hashers.LogHasher, size int64, f GetNodeFunc, expec
 	return &r, nil
 }
 
-// NewTree creates a new compact Tree with size zero. This always succeeds.
+// NewTree creates a new compact Tree with size zero.
 func NewTree(hasher hashers.LogHasher) *Tree {
-	r := Tree{
+	return &Tree{
 		hasher: hasher,
 		root:   hasher.EmptyRoot(),
 		nodes:  make([][]byte, 0),
 		size:   0,
 	}
-	return &r
 }
 
 // CurrentRoot returns the current root hash.
 func (t *Tree) CurrentRoot() []byte {
 	return t.root
 }
 
+// String describes the internal state of the compact Tree.
```

Comment:
Was this one of the things @pavelkalinnikov removed last week?

Reply:
This commit used to be "convert DumpNodes to String" and now is just "add String()", but I added use of it in the tests so it's no longer unused.  Could just delete it, but a stringer method can be helpful.

Reply:
OK.